packages/@react-aria/table/src/useTableColumnHeader.ts

```diff
@@ -80,15 +80,29 @@ export function useTableColumnHeader<T>(props: AriaTableColumnHeaderProps<T>, st
 
   let descriptionProps = useDescription(sortDescription);
 
+  let shouldDisableFocus = (state.collection.size === 0 &&
+    state.collection.body.props.loadingState !== 'loading' &&
+    state.collection.body.props.loadingState !== 'loadingMore');
+  // delay disabling focus so we get a blur event on the cell to clean up the focus ring
+  let [disableFocus, setDisableFocus]  = useState(shouldDisableFocus);
+  let focusManager = useFocusManager();
+
+  useLayoutEffect(() => {
+    if (shouldDisableFocus && !disableFocus && ref.current === document.activeElement) {
+      focusManager.focusFirst();
+    }
+    setDisableFocus(shouldDisableFocus);
+  }, [shouldDisableFocus, disableFocus, ref]);
```

Comment:
I think this can be simplified down to this:

```js
let shouldDisableFocus = state.collection.size === 0;
useEffect(() => {
  if (shouldDisableFocus && state.selectionManager.focusedKey === node.key) {
    state.selectionManager.setFocusedKey(null);
  }
}, [shouldDisableFocus, state.selectionManager, node.key]);
```

Then later on change the line to `shouldDisableFocus && {tabIndex: -1}`. Making it -1 instead of null will prevent the focus ring from getting stuck.

This should allow you to remove the extra state, `useFocusManager`, and the change in `useSelectableCollection`.